env/env.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 ThoughtWorks, Inc.
+
+// This file is part of getgauge/html-report.
+
+// getgauge/html-report is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// getgauge/html-report is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with getgauge/html-report.  If not, see <http://www.gnu.org/licenses/>.
+
+package env
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/getgauge/common"
+)
+
+const (
+	DefaultReportsDir           = "reports"
+	GaugeReportsDirEnvName      = "gauge_reports_dir" // directory where reports are generated by plugins
+	OverwriteReportsEnvProperty = "overwrite_reports"
+)
+
+func GetCurrentExecutableDir() (string, string) {
+	ex, err := os.Executable()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	target, err := os.Readlink(ex)
+	if err != nil {
+		return path.Dir(ex), filepath.Base(ex)
+	}
+	return path.Dir(target), filepath.Base(ex)
+
+}
+
+// CreateDirectory creates given directory if it doesn't exist
+func CreateDirectory(dir string) {
+	if common.DirExists(dir) {
```

Comment:
There is no need to check for directory presence as `os.MkdirAll` will do nothing and returns nil if the directory already exists.